### PR TITLE
Docs/ds 960 breadcrumbs word breaking

### PR DESCRIPTION
**Pull request by literat**
### Issue reference

https://jira.lmc.cz/browse/DS-960

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web-twig/src/Resources/components/Breadcrumbs/README.md [hunk 1]
```diff
@@ -135,5 +135,14 @@ You can add `id`, `data-*` or `aria-*` attributes to further extend the componen
 descriptiveness and accessibility. Also, UNSAFE styling props are available,
 see the [Escape hatches][escape-hatches] section in README to learn how and when to use them.
 
+### Dealing with long titles
+
+When you need to shorten the title of the BreadcrumbsItem you can use both helper class `text-truncate` or Twig's native implementation of text truncating using [`u` filter](https://twig.symfony.com/doc/3.x/filters/u.html).
+Please see the documentation for more details.
+
+```twig
+{{ 'Lorem ipsum'|u.truncate(8, '...') }}
```
Comment (crishpeen) [inline packages/web-twig/src/Resources/components/Breadcrumbs/README.md:144]:
```suggestion
{{ 'Lorem ipsum'|u.truncate(8, '…') }}
```

File: packages/web-react/src/components/Breadcrumbs/README.md [hunk 2]
```diff
@@ -91,4 +91,9 @@ Use the `BreadcrumbsItem` component for the ordered list as the component's chil
 | `UNSAFE_className` | `string`        | —               | ✕        | Wrapper custom class name                   |
 | `UNSAFE_style`     | `CSSProperties` | —               | ✕        | Wrapper custom style                        |
 
+### Dealing with long titles
+
+When you need to shorten the title of the BreadcrumbsItem you can use both helper class `text-truncate` or use any of the multiple ways in JavaScript which will truncate a string for you like `.slice` or `.substring`, e. g. `str.slice(0, num) + '...'`.
+You can also use any of the existing npm packages which deals with truncating the string.
```
Comment (crishpeen) [inline packages/web-react/src/components/Breadcrumbs/README.md:97]:
```suggestion
You can also use any of the existing npm packages which deal with truncating the string.
```

Maybe we can link the search from npm?
https://www.npmjs.com/search?q=truncate

**Review by crishpeen — COMMENTED** [review]
I see two problems:
1. The standalone class `text-truncate` does nothing without (max-)width. We need to showcase and explain how to do it. 
2. IMHO our preferred way is to use some hard truncation already built-in in Twig or JS. You probably want to have the truncation the same on all breakpoints and not split words etc.. Therefore I would start the sentence with the native part (which is currently second in your suggestions) and offer the `text-truncate` (with mention of the width demand) as a second one.

What do you think?

File: packages/web/src/scss/components/Breadcrumbs/README.md [hunk 3]
```diff
@@ -33,3 +37,20 @@
...
 ```
+
+### Dealing with long titles
+
+When you need to shorten the title of the Breadcrumbs item you can use a helper class `text-truncate`.
+
+```html
+<!-- ... --->
+<li class="d-none d-tablet-flex">
+  <svg class="Icon" width="24" height="24">
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web/src/scss/components/Breadcrumbs/README.md:48]: Removed.

File: packages/web/src/scss/components/Breadcrumbs/README.md [hunk 4]
```diff
@@ -33,3 +37,20 @@
...
+    <use xlink:href="/icons/svg/sprite.svg#chevron-right" />
+  </svg>
+  <a href="#currentUrl" aria-current="page" class="link-secondary text-truncate"
+    >This is a very long title of the current page</a
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web/src/scss/components/Breadcrumbs/README.md]: Fixed.

File: packages/web-react/src/components/Breadcrumbs/README.md (hunk 2 above)
Comment (literat) [reply packages/web-react/src/components/Breadcrumbs/README.md]: Fixed.

File: packages/web-react/src/components/Breadcrumbs/README.md [hunk 5]
```diff
@@ -91,4 +91,9 @@ Use the `BreadcrumbsItem` component for the ordered list as the component's chil
 | `UNSAFE_className` | `string`        | —               | ✕        | Wrapper custom class name                   |
 | `UNSAFE_style`     | `CSSProperties` | —               | ✕        | Wrapper custom style                        |
 
+### Dealing with long titles
+
+When you need to shorten the title of the BreadcrumbsItem you can use both helper class `text-truncate` or use any of the multiple ways in JavaScript which will truncate a string for you like `.slice` or `.substring`, e. g. `str.slice(0, num) + '...'`.
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/src/components/Breadcrumbs/README.md]: Fixed.

File: packages/web-react/src/components/Breadcrumbs/README.md [hunk 6]
```diff
@@ -91,4 +91,9 @@ Use the `BreadcrumbsItem` component for the ordered list as the component's chil
...
+You can also use any of the existing npm packages which deals with truncating the string.
+
 For detailed information see [Breadcrumbs](https://github.com/lmc-eu/spirit-design-system/blob/main/packages/web/src/scss/components/Breadcrumbs/README.md) component
```
[the remark this answers is not in the document]
Comment (literat) [reply packages/web-react/src/components/Breadcrumbs/README.md]: Done.

File: packages/web-twig/src/Resources/components/Breadcrumbs/README.md (hunk 1 above)
Comment (literat) [reply packages/web-twig/src/Resources/components/Breadcrumbs/README.md]: Fixed.

File: packages/web-twig/src/Resources/components/Breadcrumbs/README.md [hunk 7]
```diff
@@ -135,5 +135,17 @@ You can add `id`, `data-*` or `aria-*` attributes to further extend the componen
 descriptiveness and accessibility. Also, UNSAFE styling props are available,
 see the [Escape hatches][escape-hatches] section in README to learn how and when to use them.
 
+### Dealing with long titles
+
+When you need to shorten the title of the BreadcrumbsItem the preferred way is to use platform native helpers.
+Twig have implementation of text truncating using [`u` filter](https://twig.symfony.com/doc/3.x/filters/u.html).
```
Comment (crishpeen) [inline packages/web-twig/src/Resources/components/Breadcrumbs/README.md:141]:
```suggestion
Twig has an implementation of text truncating using [`u` filter](https://twig.symfony.com/doc/3.x/filters/u.html).
```

Comment (crishpeen) [reply packages/web-twig/src/Resources/components/Breadcrumbs/README.md:141]: Please the link to the bottom of the file.

File: packages/web/src/scss/components/Breadcrumbs/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (crishpeen) [reply packages/web/src/scss/components/Breadcrumbs/README.md:48]: You sure?

**Review by crishpeen — APPROVED** [review]

**Pull request merged**